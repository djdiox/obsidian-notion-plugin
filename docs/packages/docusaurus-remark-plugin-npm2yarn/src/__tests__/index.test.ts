/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import vfile from 'to-vfile';
import mdx from 'remark-mdx';
import remark from 'remark';
import npm2yarn from '../index';

const processFixture = async (name: string, options?: {sync?: boolean}) => {
  const filePath = path.join(__dirname, '__fixtures__', `${name}.md`);
  const file = await vfile.read(filePath);
  const result = await remark().use(mdx).use(npm2yarn, options).process(file);

  return result.toString();
};

describe('npm2yarn plugin', () => {
  it('works on installation file', async () => {
    const result = await processFixture('installation');

    expect(result).toMatchSnapshot();
  });

  it('works on plugin file', async () => {
    const result = await processFixture('plugin');

    expect(result).toMatchSnapshot();
  });

  it('works with sync option', async () => {
    const result = await processFixture('plugin', {sync: true});

    expect(result).toMatchSnapshot();
  });

  it('does not work when language is not set', async () => {
    const result = await processFixture('syntax-not-properly-set');

    expect(result).toMatchSnapshot();
  });

  it('does not re-import tabs components when already imported above', async () => {
    const result = await processFixture('import-tabs-above');

    expect(result).toMatchSnapshot();
  });

  it('does not re-import tabs components when already imported below', async () => {
    const result = await processFixture('import-tabs-below');

    expect(result).toMatchSnapshot();
  });
});
