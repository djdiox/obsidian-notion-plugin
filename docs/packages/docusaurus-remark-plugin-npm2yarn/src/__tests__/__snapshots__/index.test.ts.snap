// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm2yarn plugin does not re-import tabs components when already imported above 1`] = `
"import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

<Tabs>
<TabItem value="npm">

\`\`\`bash
  $ npm install --global docusaurus
\`\`\`

</TabItem>
<TabItem value="yarn" label="Yarn">

\`\`\`bash
  $ yarn add --global docusaurus
\`\`\`

</TabItem>
</Tabs>
"
`;

exports[`npm2yarn plugin does not re-import tabs components when already imported below 1`] = `
"<Tabs>
<TabItem value="npm">

\`\`\`bash
  $ npm install --global docusaurus
\`\`\`

</TabItem>
<TabItem value="yarn" label="Yarn">

\`\`\`bash
  $ yarn add --global docusaurus
\`\`\`

</TabItem>
</Tabs>

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';
"
`;

exports[`npm2yarn plugin does not work when language is not set 1`] = `
"\`\`\`npm2yarn
npm install --save docusaurus-plugin-name
\`\`\`

\`\`\`bash
npm install --save docusaurus-plugin-name
\`\`\`

\`\`\`shell
npm install --save docusaurus-plugin-name
\`\`\`
"
`;

exports[`npm2yarn plugin works on installation file 1`] = `
"import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>
<TabItem value="npm">

\`\`\`bash
  $ npm install --global docusaurus
\`\`\`

</TabItem>
<TabItem value="yarn" label="Yarn">

\`\`\`bash
  $ yarn add --global docusaurus
\`\`\`

</TabItem>
</Tabs>
"
`;

exports[`npm2yarn plugin works on plugin file 1`] = `
"import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Installing a plugin

A plugin is usually a npm package, so you install them like other npm packages using npm.

<Tabs>
<TabItem value="npm">

\`\`\`bash
npm install --save docusaurus-plugin-name
\`\`\`

</TabItem>
<TabItem value="yarn" label="Yarn">

\`\`\`bash
yarn add docusaurus-plugin-name
\`\`\`

</TabItem>
</Tabs>
"
`;

exports[`npm2yarn plugin works with sync option 1`] = `
"import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Installing a plugin

A plugin is usually a npm package, so you install them like other npm packages using npm.

<Tabs groupId="npm2yarn">
<TabItem value="npm">

\`\`\`bash
npm install --save docusaurus-plugin-name
\`\`\`

</TabItem>
<TabItem value="yarn" label="Yarn">

\`\`\`bash
yarn add docusaurus-plugin-name
\`\`\`

</TabItem>
</Tabs>
"
`;
